src/Npgsql/NpgsqlCommandBuilder.cs

```diff
@@ -122,15 +122,12 @@ public override string QuoteSuffix
         /// <param name="command">NpgsqlCommand whose function parameters will be obtained.</param>
         public static void DeriveParameters(NpgsqlCommand command)
```

Comment:
I agree  that splitting up the code for `DeriveParametersForFunction()` and `DeriveParametersForQuery()`  between `NpgsqlCommand` and `NpgsqlCommandBuilder` doesn't make sense.
I made it  callable on an instance of NpgsqlCommand directly, but kept it internal as I don't think that we should confuse developers by having two public interface methods that do the same thing and I want to maintain iinterface compatibility with other .Net data providers.

Reply:
OK with me to leave the current situation where derivation is only publicly callable from `NpgsqlCommandBuilder` and not from `NpgsqlCommand`.

Regarding where the actual code should live, I'd say as before that it should be in NpgsqlCommand, making use of partial class to prevent the file from growing too big.

Reply:
I think putting it into a separate file and making the class partial is a matter of taste.
Personally I think the partial feature is a little overused nowadays.
Originally the feature was added to allow part of the code to be autogenerated and add code without losing it on regeneration. I find it quite non-obvious that the implementation of a class might be distributed amongst several classes.
I tend to do Ctrl+M whenever I open a C# file in VS and only expand the region I'm interested in but that obviously doesn't work in notepad.
If you'd like to have it in a separate fileI'll move it.

Reply:
It's OK, as you say it's a matter of taste, I'm fine with leaving it in a region as-is.